### ISSUE TITLE
Feature Request: Generate Basic HTML Wrapper

### ISSUE THREAD

**Issue by renaun** [RandoriAS/randori-compiler#55]
Have a flag to create a basic html wrapper for application using randori-compiler that are not using randori framework. The index template could be in the sdk repo and changable.

So by default it looks for index.template.html in the randori-sdk or has a override flag.

Create wrapper and use default:
-html-wrapper 
Create wrapper based on template.html:
-html-wrapper=/path/to/template.html

**Comment by teotigraphix** [comment]
@labriola can you post here when you get this in the repo so I would know where to look. This will also be a category in the Bundle when the framework is archived. So I envision

```
randori-sdk/randori-framework/html/template/index.template.html
```

I need that file structure for it to be easy to add.

**Comment by renaun** [comment]
I'll give an example of what goes into the template.html once we have worked that with the updates @labriola is doing

**Comment by labriola** [comment]
I will propose back some thoughts here when the next release is done. We should make sure we mirror what we are thinking about in the plugint oo

**Comment by labriola** [comment]
We need to be cautious though. What a Randori project needs is different than what Renaun needs presently. So, i urge caution until we think this through

**Comment by teotigraphix** [comment]
@labriola  Seeing as the Bundle API is completely flexible, this is working out exactly the way I had envisioned with libraries/container/category IE randori-framework/html/template.

I hope you can see this now Mike. :)

As far as the plugin, following the above guideline, it will be as easy as it is for them copying the js right now

**Comment by labriola** [comment]
@teotigraphix that would imply a different SDK for Renaun's use and people building Randori apps. Perhaps thats what we do. I am just urging discussion first

**Comment by teotigraphix** [comment]
@labriola I only follow, I assume you are talking about what goes in the html template? I have nothing to say about this as you know.

**Comment by teotigraphix** [comment]
@labriola Ah you mean having an html template that is configurable? Randori doesn't need that is what you are saying?

**Comment by renaun** [comment]
Well if its just guice needed for bootstrapping the non-mvc app isn't still part of the sdk?

Maybe I don't understand Randori all the way but the examples I have seen show a basic html doc with:
    <script type="text/javascript">

```
    //Start the randori framework
    $(function () {
        var randoriApp = new randori.startup.RandoriBootstrap(document);

        //Passing a true to launch enables debug mode where all request attempt to disallow caching
        randoriApp.launch( true );
    });
</script>
```

So a only guice one might be just that basic html file with a simple one/two liner bootstrapper that loads the main class of the app. 

```
<script type="text/javascript">
   var app = Guice.loadClass("myApp"); // myApp then knows its constructor is like main
</script>
```

That does imply knowledge of the app name which is not needed for Randori i think.

**Comment by labriola** [comment]
@teotigraphix I am just saying Renaun's use case is cool, it just wasnt the one we were building for so I want to support his and the app case. Just want to do so elegantly.

@renaun I agree, although we might want to defer the Guice call even as right now that will happen on browser start before the DOM is ready which may throw people for a looop

**Comment by renaun** [comment]
@labriola agreed and that can be part of the template so ppl get an out of the box compile -> run way of doing things. Then they can go back and add/change what they need specific to their app.
